UI refactor

tchassijordan opened the pull request:
### Major Changes
- Refactored the UI of the Sidebar Component. The old UI was too plain, UI design fundamentals weren't respected. So I refactored all those out.
- Broke down the components of the app into separate reusable components with each having a single responsibility.
- Refactored source code

###Preview Change:
 
![Screenshot from 2022-04-23 02-39-26](https://user-images.githubusercontent.com/76742818/164861575-76bf480d-c644-4c77-b8db-a6b91296ca34.png)

tchassijordan committed:
refactored entire app.

tchassijordan committed:
added icons to sidebar and code refacro
r

fdrissi commented on components/NavbarItem.tsx:
```diff
@@ -0,0 +1,24 @@
+  cls?: string;
+};
+
+export default function NavbarItem({
```
SidbarItem will be a better naming

fdrissi commented on components/Sidebar.tsx:
```diff
@@ -1,11 +1,14 @@
-        <li className="hover:text-stone-400 cursor-pointer border-b-2 border-stone-400">Contact</li>
-      </ul>
+    <div className="w-64 bg-[#222628] text-white flex flex-col border-[1px] border-[#a5a19950]">
+      <h1 className="px-5 py-4 mb-3 font-bold text-lg tracking-wide cursor-pointer border-b-[1px] border-[#a5a19950]">Quick Links</h1>
+      {children}
```
Move the title to the props, make it optional, and set the default value to `Quick Links`

fdrissi commented on components/Sidebar.tsx:
```diff
@@ -1,11 +1,14 @@
-        <li className="hover:text-stone-400 cursor-pointer border-b-2 border-stone-400">About</li>
-        <li className="hover:text-stone-400 cursor-pointer border-b-2 border-stone-400">Contact</li>
-      </ul>
+    <div className="w-64 bg-[#222628] text-white flex flex-col border-[1px] border-[#a5a19950]">
```
border-[1px] === border.
bg-[#222628] this maybe will be your primary color, and this border-[#a5a19950] maybe your primary border color, so add them to tw theme.

fdrissi commented on components/Table.tsx:
```diff
@@ -0,0 +1,51 @@
+import "ag-grid-enterprise";
+import "ag-grid-community/dist/styles/ag-grid.css";
+import "ag-grid-community/dist/styles/ag-theme-alpine-dark.css";
+
+export default function Table():ReactElement {
+  //define the column header for our table in a local state
+  const [columnDefs] = useState([
```
This is not a reusable component.

fdrissi commented on pages/index.tsx:
```diff
@@ -1,65 +1,24 @@
+          <NavbarItem to={"/"}>country</NavbarItem>
+          <NavbarItem to={"/"} cls="opacity-60">sectors</NavbarItem>
+          <NavbarItem to={"/"} cls="opacity-60">signed amount</NavbarItem>
+          <NavbarItem to={"/"} cls="opacity-60">date</NavbarItem>
```
move these links to an array, and handle the active link style dynamically from inside `NavbarItem` component

fdrissi reviewed (commented):
This is much better, you made good progress, good job.
don't forget to work also on commit messages

tchassijordan committed:
refactor: renamed the fillProp in each icon component to fillColor

fillProp was the name given to the dynamic color being sent to an icon component, I renamed it to fillColor to improve readability

tchassijordan committed:
Renamed NavbarItem to SidebarItem for better naming convention
Moved Table component out of the components folder back into pages/index.tsx because it is not a reusable component
Modified the Home component in pages so that it passes icons and menu items to the new SidebarItem dynamically
Customized tailwindcss file to include  custom color palette
Modified Sidebar component so that it can receive Sidebar title dynamically

tchassijordan commented:
@fdrissi Thanks so much for the review. 
I have improved the code base following your comments.
